cookbooks/aws-parallelcluster-environment/recipes/init/restore_home_shared_data.rb

```diff
@@ -18,19 +18,65 @@
   # Restore the shared storage home data if it doesn't already exist
   # This is necessary to preserve any data in these directories that was
   # generated during the build of ParallelCluster AMIs after converting to
-  # shared storage and backed up to a temporary location previously
+  # shared storage and backed up to a temporary location previously.
   # Before removing the backup, ensure the data in the new home is the same
   # as the original to avoid any data loss or inconsistency. This is done
   # by using rsync to copy the data and diff to check for differences.
+  # The diff command excludes files that were originally in /home to focus on
+  # the newly synchronized files. This approach is necessary because the /home
+  # directory may contain pre-existing files such as slurm-*.out generated by
+  # running SLURM jobs and the automatically created lost+found directory.
   # Remove the backup after the copy is done and the data integrity is verified.
   bash "Restore /home" do
     user 'root'
     group 'root'
     code <<-EOH
+      # Generate a list of existing files and dirs in /home before the sync
+      find /home -mindepth 1 > /tmp/home_existing_files.txt
+
+      # Initialize an empty set for exclude options and directories to exclude
+      exclude_options=""
+      declare -A exclude_dirs
+
+      # Process each file in the list
+      while IFS= read -r file; do
+        # Remove the /home/ prefix
+        relative_path=${file#/home/}
```

Comment:
FYI there is a linux command to get the relative path: `realpath `

See example of usage: https://stackoverflow.com/questions/2564634/convert-absolute-path-into-relative-path-given-a-current-directory-using-bash

Reply:
In this case, it's simpler and more efficient to manipulate the string directly, so why don't we keep the existing logic.

Reply:
sure, it was just for knowledge sharing (see FYI). I think it's useful to take the chance of a PR to share insights on alternative approaches. Either ways are fine.